public/js/controllers/market_controller.js

```diff
@@ -1128,32 +1150,42 @@ export default class extends Controller {
     btn.classList.add('btn-selected')
     aggStacking = btn.name === 'on'
     this.graph.updateOptions({ stackedGraph: aggStacking, fillGraph: aggStacking })
+    settings.stack = btn.name
+    this.updateQueryString()
   }
 
   setZoom (e) {
     var btn = e.target || e.srcElement
     if (btn.nodeName !== 'BUTTON' || !this.graph) return
     this.setZoomPct(parseInt(btn.name))
     var stats = this.graph.getOption('stats')
-    var spread = stats.midGap * parseFloat(btn.name) / 100
-    this.graph.updateOptions({ dateWindow: [stats.midGap - spread, stats.midGap + spread] })
+    if (stats) {
+      var spread = stats.midGap * parseFloat(btn.name) / 100
+      this.graph.updateOptions({ dateWindow: [stats.midGap - spread, stats.midGap + spread] })
+    }
+    settings.zoom = parseInt(btn.name)
```

Comment:
Can we set a `data-percentage` on the buttons rather than using the `name` attribute for this purpose?

Reply:
Done.